src/lib/y2storage/boot_requirements_checker.rb

```diff
@@ -61,26 +67,36 @@ def needed_partitions(target = :desired)
 
   protected
 
-    attr_reader :settings
+    # @return [Devicegraph] starting situation.
     attr_reader :devicegraph
 
+    # @return [Array<Planned::Device>] devices that are already planned to be
+    #   added to the starting devicegraph.
+    attr_reader :planned_devices
+
+    # @return [String, nil] device name of the disk that the system will try to
+    #   boot first. Only useful in some scenarios like legacy boot.
+    attr_reader :boot_disk_name
+
     def arch
       @arch ||= StorageManager.instance.arch
     end
 
     def strategy
       return @strategy unless @strategy.nil?
 
-      if arch.x86? && arch.efiboot?
-        @strategy = BootRequirementsStrategies::UEFI.new(settings, devicegraph)
-      elsif arch.s390?
-        @strategy = BootRequirementsStrategies::ZIPL.new(settings, devicegraph)
-      elsif arch.ppc?
-        @strategy = BootRequirementsStrategies::PReP.new(settings, devicegraph)
-      end
-
-      # Fallback to Legacy as default
-      @strategy ||= BootRequirementsStrategies::Legacy.new(settings, devicegraph)
+      klass =
+        if arch.x86? && arch.efiboot?
```

Comment:
is there any reason to use libstorage-arch instead of yast2 Arch class? I do not see it documented at https://github.com/ancorgs/yast-storage-ng/blob/272abb79b527f2bf0e869f74b029c81ac68936fa/src/lib/y2storage/storage_manager.rb